[Swati Rajesh] iP

swatirajesh277 opened the pull request:
### **⋆˚࿔ CHERRYBOT 𝜗𝜚˚⋆**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

CherryBot is here to lighten your mental load. No more worrying about what needs to be done—just focus on your big ideas, and let CherryBot handle the rest. 🍒

**Key features :**
1. Easy Task Management: Quickly add and organize your tasks. ✅
2. Super Fast: Get things done in a snap. ⚡️
3. Completely Free: No costs, just pure efficiency. 🎉

**How to get started :**
1. Download CherryBot – it’s a quick [click](https://github.com/swatirajesh277/ip) away. 📥
2. Open it up – CherryBot is ready to work! 💻
3. Add Your Tasks – type them in and let CherryBot handle the rest. 📝
4. Let CherryBot Do Its Magic – your to-do list will be sorted in no time. ✨

**Coming soon :**
- [x] Managing tasks
- [x] Searching tasks
- [x] Managing deadlines
- [ ] Easy to use and impressive UI _(coming soon)_
- [ ] Sorting tasks _(coming soon)_

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

swatirajesh277 committed:
replace name and add bot intro

swatirajesh277 committed:
add echo except bye

swatirajesh277 committed:
add support for list

swatirajesh277 committed:
add list

swatirajesh277 committed:
add task support

swatirajesh277 committed:
add mark unmark task

swatirajesh277 committed:
add todo event deadline

swatirajesh277 committed:
add complete implementation of todo event deadline

swatirajesh277 committed:
ui testing bug fix

swatirajesh277 committed:
add exception

swatirajesh277 committed:
add delete

swatirajesh277 committed:
add text file

swatirajesh277 committed:
edit file according to list

swatirajesh277 committed:
add date and time

swatirajesh277 committed:
add ui class

swatirajesh277 committed:
add storage class

swatirajesh277 committed:
add command

swatirajesh277 committed:
add more oop

swatirajesh277 committed:
Merge branch 'A-moreOOP'

swatirajesh277 committed:
add packages

swatirajesh277 committed:
Merge branch 'A-Packages'

nlqhung130403 commented on src/main/java/cherrybot/ui/CherryBot.java:
```diff
@@ -0,0 +1,51 @@
+
+public class CherryBot {
+    private Storage storage;
+    private TaskList tasks;
```
It's nice that you use plural noun for collection object 👍

nlqhung130403 commented on src/main/java/cherrybot/ui/Parser.java:
```diff
@@ -0,0 +1,97 @@
+
+public class Parser {
+
+    public static Command parse(String msg) throws CherryBotException {
```
You handle most of the exceptions very nicely 👌

nlqhung130403 commented on src/main/java/cherrybot/ui/Parser.java:
```diff
@@ -0,0 +1,97 @@
+            String activity = msg.trim().substring(6);
+
+            if (!activity.contains(" /from") && !activity.contains(" /to")) {
+                throw new CherryBotException("Event command invalid : <description> /from <time> /to <time>");
```
I like how you give the command format to aid users

nlqhung130403 commented on src/main/java/cherrybot/ui/Ui.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    public static String readCommand() {
+        Scanner scanner = new Scanner(System.in);
```
I think that you should close the canner once you finish using it.

swatirajesh277 committed:
fix bugs

swatirajesh277 committed:
Merge branch 'add-gradle-support'

swatirajesh277 committed:
add gradle support

swatirajesh277 committed:
fix gradle bugs

swatirajesh277 committed:
fix gradle errors

swatirajesh277 committed:
add new test file

swatirajesh277 committed:
fix bug

swatirajesh277 committed:
add many junit tests, fix bugs

swatirajesh277 committed:
Merge branch 'A-JUnit'

swatirajesh277 committed:
add jar support

swatirajesh277 committed:
add javadoc comments

swatirajesh277 committed:
add find command

swatirajesh277 committed:
Merge branch 'branch-A-JavaDoc'

swatirajesh277 committed:
Merge branch 'branch-Level-9'

swatirajesh277 committed:
fix bugs

swatirajesh277 committed:
add ui elements

swatirajesh277 committed:
fix ui bugs

swatirajesh277 committed:
add ui working component

swatirajesh277 committed:
Merge branch 'branch-Level-10'

swatirajesh277 committed:
fix ui

swatirajesh277 committed:
Blank space to write commands

A blank space without any prompts that message and commands should be typed there is not user friendly as users cannot be expected to automatically assume where ti type their commands

Let's
-update the ui for mainwindow to contain a place holder text reading "type message here..."

swatirajesh277 committed:
add new data

swatirajesh277 committed:
Parse command : add assert statements

No assert statements in parse command

Assert statements help with debugging of code and catching internal errors.

Let's,
* update the parse command to have some assert methods

swatirajesh277 committed:
Merge pull request #2 from swatirajesh277/branch-A-Assertions

Parse command : add assert statements

swatirajesh277 committed:
solve code quality issues

swatirajesh277 committed:
Merge pull request #3 from swatirajesh277/branch-A-CodeQuality

solve code quality issues

swatirajesh277 committed:
Merge branch 'master' of https://github.com/swatirajesh277/ip

swatirajesh277 committed:
Add a sort command

No sort command to sort tasks according to deadline

Users can sort tasks and know which task is more important and needs to get done.

Let's,
* update the parser to identify sort
* add a sort command by making use of comparator

swatirajesh277 committed:
Merge branch 'branch-BCD-extensions'

swatirajesh277 committed:
Making ui more user friendly

Unnecessary ui components

User friendly ui makes the experience of the application better

Let's,
* update the dividers after each statement
* have the application name on the window of the chatbot

swatirajesh277 committed:
command: make them case-insensitive

All commands are case insensitive

A case-insensitive command is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

Let's,
* update the search algorithm to use case-insensitive commands
* add a list of commands in intro message to help users

swatirajesh277 committed:
fix bugs

swatirajesh277 committed:
add remaining javadoc

swatirajesh277 committed:
fix some javadoc spacing

swatirajesh277 committed:
update readme, add ui.png

swatirajesh277 committed:
update readme

swatirajesh277 committed:
update readme to make it readable

swatirajesh277 committed:
fix bugs

swatirajesh277 committed:
fix hardcode file

swatirajesh277 committed:
fix file reading bugs

swatirajesh277 committed:
Fix bugs in Readme
